utility/actor.go

```diff
@@ -149,11 +149,11 @@ func (u *UtilityContext) GetMaxPausedBlocks(actorType coreTypes.ActorType) (maxP
 		return 0, typesUtil.ErrUnknownActorType(actorType.String())
 	}
 
-	var er error
```

Comment:
This is the only major issue I'm seeing in this PR.

I realize that part of the motivation was to remove `GetIntParam` and the other 2 functions, but now the onus of type casting is on the caller of `GetParamter`, which doesn't validate/check the cast at all.

What if we change `GetParameter` to accept a generic `[int | string | []byte]` and let the called specify it explicitly, so an error is returned if the cast fails?

Reply:
@Olshansk I tried this but interface functions cannot have type parameters - this I thought would be the best work around, next to having a type casted wrapper for `GetParameter()` but this seems like a step backwards to what there was before.

Unless we have something like 
```go
func GetParam[T int | string | []byte](paramName string, height int64) (T, error)
```

Reply:
I don't have an immediate answer to this. It's one of the reasons I needed to think about #436 and search through the codebase to make sure no better/duplicate solutions exist.

I think you're getting pretty familiar with the layout, so I have two follow-up questions:

1. Does this duplicate work with  `func getParamOrFlag[T int | string | []byte]` in `gov.go`? Is there an opportunity to have logic consolidation there?

2. Do you think it makes sense to have the caller of `GetParam` be forced to type cast and check for errors?

Again, this isn't a loaded question, I just haven't had time to sit and go through all the flows to see what the cleanest path is.

Reply:
So the `GetParam` function is probably a bad idea - I have just push a commit that removes the need for type casting by relying on the same helper functions in the utility context (now with added type casting checks) these are wrappers around `GetParameter()` returning solid types

Reply:
@Olshansk regarding point 1 this would have probably just been a duplication of this function which is already wrapped by `GetParameter()` it would be better to avoid type casting and type parameter usages by the end user. However I do believe that whenever someone is using the parameters they will know the type they aim to be using - so the `utility/gov.go` helper functions can easily wrap the `GetParameter()` call handle any errors around type casting and shoot out the type they need at the current height

Reply:
Quick thought: Change the `pokt:"val_type=..."` tags in the Protobuf to reflect the type the parameter actually is. This allows for the same method of encoding the data (`*big.Int`=`string`, `[]byte`=`string`, etc.). Currently the `pokt:"val_type=BIGINT"` refers to both `int64` and `*big.Int` (`persistence/types/gov.go`)

![image](https://user-images.githubusercontent.com/53987565/214920237-8de33174-3da0-4f63-8f3a-c3afee9c4eb6.png)

This would also mean the constants above would be changed to reflect the type in the switch statement - not altering any encoding but helping the parsing - I may be wrong here but I don't see a benefit of having a `*big.Int` val type be string and also a `[]byte` val type be a string and an `int64` val type be a `*big.Int`?

However this means the `GetParameter()` function could leverage `govParamMetadataMap` map built in `persistence/types/gov.go` (instead of using its own reflection of the functions return values) and accurately return the correct type of the parameter - then we would simply need the wrappers to do typecasting + error checking on this.

Whenever we return a generic and want to pass this into another function there must be a type-casting step, unless we allow the receiving functions to accept generics which is probably a bad idea. So with a generic function we should at least have it handle the type conversions (all parameters are stored as strings and the `GetParameter()` function only actually handles the `int` and `string` types due to its inability to accurately detect types) and any errors that may come from this

Reply:
@h5law

**Side context**: I'm in the process of refactoring the utility module (#476) and revisiting (in the future) the entire `string/byte/BigInt` conversions is something I'd like to do. It'll require both thought and discussion, so I don't want that to be a blocker/distraction right now though.

The PR, as it stands right now, since to be a slight code health improvement.

Regarding your idea - I think it's a potential approach but have lost some context, and just don't see it as an immediate need/priority. What if we just create a ticket for the M5 milestone and revisit it then?

---

If you think this is something that should be done now though, to help/ease development, happy to discuss further.

Reply:
Yea, I'd prefer we close it out and leave it as a reference in the ticket you created.

I think it requires a bit more thought on what we're trying to achieve here, which will become clearer once utility matures a bit.